js/common/bootstrap/modules/buffer.js

```diff
@@ -967,7 +967,10 @@ global.DEFINE_MODULE('buffer', (function () {
       return Buffer.from(subject[Symbol.toPrimitive](), offsetOrEncoding, length);
     }
     if (typeof subject.valueOf === 'function') {
-      return Buffer.from(subject.valueOf(), offsetOrEncoding, length);
+      const value = subject.valueOf();
+      if (value !== subject) {
```

Comment:
Not performing this check causes infinite recursion. The only false positive is `NaN`, which is already eliminated by the `typeof` check earlier.